app/models/admin_session.rb

```diff
@@ -5,6 +5,10 @@ class AdminSession < DfeSignIn::AuthenticatedSession
   SUPPORT_AGENT_DFE_SIGN_IN_ROLE_CODE = "teacher_payments_support"
   PAYROLL_OPERATOR_DFE_SIGN_IN_ROLE_CODE = "teacher_payments_payroll"
 
+  def user
+    @user ||= DfeSignIn::User.find_or_create_by(dfe_sign_in_id: user_id)
```

Comment:
Didn't we discuss doing this at the point of user sign-in and storing our own UID as the session key?